core/dbt/contracts/graph/manifest.py

```diff
@@ -936,6 +938,31 @@ class Manifest(MacroMethods, dbtClassMixin):
         metadata={"serialize": lambda x: None, "deserialize": lambda x: None},
     )
 
+    # This and the following methods can be refacong into a mixin
+    # and we should do so next time we do it for another artifact.
+    @classmethod
+    def validate(cls, data: Any) -> None:
```

Comment:
I think this should be on the `WritableManifest` class in dbt/artifacts, not the `Manifest` internal object, since it is defining behaviour related to ser/de which dbt/artifacts should be responsible for. This will also make it easier to extend the pattern to other artifact schemas (e.g. making a `ExtensibleArtifactMixin`, or just making `ArtifactMixin` have this `validate` implementation)